RealmSwift/List.swift

```diff
@@ -0,0 +1,298 @@
+
+    :param: object An object.
+    */
+    public func append(object: T) {
+        _rlmArray.addObject(unsafeBitCast(object, RLMObject.self))
+    }
+
+    /**
+    Appends the objects in the given sequence to the end of the list.
+
+    :warning: This method can only be called during a write transaction.
+
+    :param: objects A sequence of objects.
+    */
+    public func extend<S: SequenceType where S.Generator.Element == T>(objects: S) {
+        for obj in SequenceOf<T>(objects) {
+            _rlmArray.addObject(unsafeBitCast(obj, RLMObject.self))
+        }
+    }
+
+    /**
+    Inserts the given object at the given index.
+
+    :warning: This method can only be called during a write transaction.
+    :warning: Throws an exception when called with an index smaller than zero or greater than 
+              or equal to the number of objects in the list.
+
+    :param: object An object.
+    :param: index  The index at which to insert the object.
+    */
+    public func insert(object: T, atIndex index: Int) {
+        throwForNegativeIndex(index)
+        _rlmArray.insertObject(unsafeBitCast(object, RLMObject.self), atIndex: UInt(index))
+    }
+
+    /**
+    Removes the object at the given index from the list. Does not remove the object from the Realm.
+
+    :warning: This method can only be called during a write transaction.
+    :warning: Throws an exception when called with an index smaller than zero or greater than
+              or equal to the number of objects in the list.
+
+    :param: index The index at which to remove the object.
+    */
+    public func removeAtIndex(index: Int) {
+        throwForNegativeIndex(index)
+        _rlmArray.removeObjectAtIndex(UInt(index))
+    }
+
+    /**
+    Removes the last object in the list. Does not remove the object from the Realm.
+
+    :warning: This method can only be called during a write transaction.
+    */
+    public func removeLast() {
+        _rlmArray.removeLastObject()
+    }
+
+    /**
+    Removes all objects from the List. Does not remove the objects from the Realm.
+
+    :warning: This method can only be called during a write transaction.
+    */
+    public func removeAll() {
+        _rlmArray.removeAllObjects()
+    }
+
+    /**
+    Replaces an object at the given index with a new object.
+
+    :warning: This method can only be called during a write transaction.
+    :warning: Throws an exception when called with an index smaller than zero or greater than
+              or equal to the number of objects in the list.
+
+    :param: index  The list index of the object to be replaced.
+    :param: object An object to replace at the specified index.
+    */
+    public func replace(index: Int, object: T) {
+        throwForNegativeIndex(index)
+        _rlmArray.replaceObjectAtIndex(UInt(index), withObject: unsafeBitCast(object, RLMObject.self))
+    }
+}
+
+extension List : ExtensibleCollectionType {
```

Comment:
we've kept the colon next to the type name for all our other declarations and should also do it here: `extension List : ExtensibleCollectionType`.